app/lib/backend/api_requests/api/shared.dart

```diff
@@ -10,7 +10,11 @@ import 'package:instabug_http_client/instabug_http_client.dart';
 import 'package:internet_connection_checker_plus/internet_connection_checker_plus.dart';
 
 Future<String> getAuthHeader() async {
-  if (SharedPreferencesUtil().authToken == '') {
+  DateTime? expiry = DateTime.fromMillisecondsSinceEpoch(SharedPreferencesUtil().tokenExpirationTime);
+  if (SharedPreferencesUtil().authToken == '' ||
+      expiry.isBefore(DateTime.now()) ||
+      expiry.isAtSameMomentAs(DateTime.fromMillisecondsSinceEpoch(0)) ||
+      (expiry.isBefore(DateTime.now().add(const Duration(minutes: 5))) && expiry.isAfter(DateTime.now()))) {
```

Comment:
What happens if the code is invalid, do we send the user to auth again or something?

Reply:
If the token is invalid or expired, getIdToken() is called which will issue a new token. The newly issued token is updated in shared prefs (previously it wasn't being done, and we were only checking `SharedPreferencesUtil().authToken == ''` while calling `getAuthHeader()`, because of which we were getting expired tokens). The newly issued token is then used for all requests.

Reply:
I see, so it should never fail

Reply:
Yes ideally it should not unless the refresh token is revoked while the user is using the app (it can only be done by deleting the account or from the backend using the `revokeToken` method)